sklearn/calibration.py

```diff
@@ -202,11 +187,42 @@ def fit(self, X, y, sample_weight=None):
             base_estimator = self.base_estimator
 
         if self.cv == "prefit":
+            check_is_fitted(self.base_estimator)
+            # Set `n_features_in_` attribute
+            try:
+                self.n_features_in_ = self.base_estimator.n_features_in_
```

Comment:
Since it would happen only for some specific estimator we can try (like `DictVectorizer`):

```python
if hasattr(self, "n_features_in_):
    self.n_features_in_ = self.base_estimator.n_features_in_
```